core/src/smartcontracts/isi/domain.rs

```diff
@@ -82,31 +52,9 @@ pub mod isi {
             self,
             authority: <Account as Identifiable>::Id,
             wsv: &WorldStateView<W>,
-        ) -> Result<Vec<DataEvent>, Self::Error> {
-            let asset_definition = self.object.clone();
-            asset_definition
-                .id
-                .name
-                .validate_len(wsv.config.ident_length_limits)
-                .map_err(Error::Validate)?;
-            let domain_id = asset_definition.id.domain_id.clone();
-            let mut domain = wsv.domain_mut(&domain_id)?;
-            match domain.asset_definitions.entry(asset_definition.id.clone()) {
-                Entry::Vacant(entry) => {
-                    let _ = entry.insert(AssetDefinitionEntry {
-                        definition: asset_definition,
-                        registered_by: authority,
-                    });
-                }
-                Entry::Occupied(entry) => {
-                    return Err(Error::Repetition(
-                        InstructionType::Register,
-                        IdBox::AccountId(entry.get().registered_by.clone()),
-                    ))
-                }
-            }
-
-            Ok(vec![DataEvent::new(self.object.id, DataStatus::Created)])
+        ) -> Result<(), Self::Error> {
+            let asset_definition = self.object;
+            wsv.register(asset_definition, authority)
```

Comment:
as far as I can see this is the only instruction for which `authority` is actually used, but it pollutes the `Execute::execute` API for all instructions. Do you think it would make sense to make `authority` part of the `AssetDefinition` structure?

Reply:
I agree. Although I think that `authority` shall be part of more ISI when we add more permissions, I don't see why it couldn't be part of the instruction itself, and be validated in tandem. I would make it in a separate PR, though.